packages/esm-outpatient-app/src/patient-info/appointment-details.component.tsx

```diff
@@ -0,0 +1,194 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import dayjs from 'dayjs';
+import { InlineLoading } from 'carbon-components-react';
+import { useAppointments } from './appointments.resource';
+import styles from './appointment-details.component.scss';
+import { formatDatetime, parseDate } from '@openmrs/esm-framework';
+
+interface AppointmentDetailsProps {
+  patientUuid: string;
+}
+
+interface PastAppointmentDetailsProps {
+  pastAppointments: Array<any>;
+}
+
+interface UpcomingAppointmentDetailsProps {
+  upcomingAppointments: Array<any>;
+}
+
+const PastAppointmentDetails: React.FC<PastAppointmentDetailsProps> = ({ pastAppointments }) => {
+  const { t } = useTranslation();
+
+  return (
+    <div className={styles.widgetCard}>
+      {pastAppointments.length >= 1 ? (
+        pastAppointments.map((appointment, index) => {
+          return (
+            <>
+              <p className={styles.title}>{t('lastEncounter', 'Last encounter')}</p>
+              <p className={styles.subtitle}>
+                {formatDatetime(parseDate(appointment.startDateTime))} · {appointment.service.name} ·{' '}
+                {appointment.location.name}{' '}
+              </p>
+            </>
+          );
+        })
+      ) : (
+        <p className={styles.content}>
+          {t('noLastEncounter', 'There is no last encounter to display for this patient')}
+        </p>
+      )}
+    </div>
+  );
+};
+
+const UpcomingAppointmentDetails: React.FC<UpcomingAppointmentDetailsProps> = ({ upcomingAppointments }) => {
+  const { t } = useTranslation();
+
+  return (
+    <div className={styles.widgetCard}>
+      {upcomingAppointments.length >= 1 ? (
+        upcomingAppointments.map((appointment, index) => {
+          return (
+            <>
+              <p className={styles.title}>{t('returnDate', 'Return date')}</p>
+              <p className={styles.subtitle}>
+                {formatDatetime(parseDate(appointment.startDateTime))} · {appointment.service.name} ·{' '}
+                {appointment.location.name}{' '}
+              </p>
+            </>
+          );
+        })
+      ) : (
+        <p className={styles.content}>{t('noReturnDate', 'There is no return date to display for this patient')}</p>
+      )}
+    </div>
+  );
+};
+
+const AppointmentDetails: React.FC<AppointmentDetailsProps> = ({ patientUuid }) => {
+  const { t } = useTranslation();
+  const headerTitle = t('appointments', 'Appointments');
+  const startDate = dayjs(new Date().toISOString()).subtract(6, 'month').toISOString();
+  const {
+    // data: appointmentsData,
+    isError,
+    isLoading,
+    isValidating,
+  } = useAppointments(patientUuid, startDate, new AbortController());
+
+  let appointmentsData = {
```

Comment:
Mock data should not be kept in the component file.

Reply:
yes, i am waiting for the endpoint

Reply:
I think I already saw these changes last week. Thanks for moving the mock data out of the component.